contracts/OlasMech.sol

```diff
@@ -90,118 +87,91 @@ abstract contract OlasMech is Mech, IErrorsMech, ImmutableStorage {
     }
 
     /// @dev Performs actions before the delivery of a request.
-    /// @param requester Requester address.
     /// @param requestId Request Id.
     /// @param data Self-descriptive opaque data-blob.
     /// @return requestData Data for the request processing.
-    function _preDeliver(
-        address requester,
-        uint256 requestId,
-        bytes memory data
-    ) internal virtual returns (bytes memory requestData);
+    function _preDeliver(uint256 requestId, bytes memory data) internal virtual returns (bytes memory requestData);
 
     /// @dev Registers a request.
     /// @param requester Requester address.
-    /// @param requestId Request Id.
-    /// @param data Self-descriptive opaque data-blob.
+    /// @param requestIds Set of request Ids.
+    /// @param datas Set of corresponding self-descriptive opaque data-blobs.
     function _request(
         address requester,
-        uint256 requestId,
-        bytes memory data
+        uint256[] memory requestIds,
+        bytes[] memory datas
     ) internal virtual {
-        mapUndeliveredRequestsCounts[requester]++;
-        // Record the requestId => sender correspondence
-        mapRequestAddresses[requestId] = requester;
```

Comment:
As specified above, no need to track requesters by mechs, all this info is in the MM.